java/ql/src/semmle/code/java/Annotation.qll

```diff
@@ -44,12 +44,65 @@ class Annotation extends @annotation, Expr {
     result = this.getType().getAnnotationElement(name)
   }
 
-  /** Gets a value of an annotation element. */
+  /**
+   * Gets a value of an annotation element. This includes default values in case
+   * no explicit value is specified.
+   */
   Expr getAValue() { filteredAnnotValue(this, _, result) }
 
-  /** Gets the value of the annotation element with the specified `name`. */
+  /**
+   * Gets the value of the annotation element with the specified `name`.
+   * This includes default values in case no explicit value is specified.
+   */
   Expr getValue(string name) { filteredAnnotValue(this, this.getAnnotationElement(name), result) }
 
+  /**
+   * If the value type of the annotation element with the specified `name` is an enum type,
+   * gets the enum constant used as value for that element. This includes default values in
+   * case no explicit value is specified.
+   */
+  EnumConstant getValueEnumConstant(string name) { result = getValue(name).(FieldRead).getField() }
+
+  /**
+   * If the value type of the annotation element with the specified `name` is `String`,
+   * gets the string value used for that element. This includes default values in case no
+   * explicit value is specified.
+   */
+  string getValueString(string name) {
+    // Uses CompileTimeConstantExpr instead of StringLiteral because value can
+    // be read of final variable as well
+    result = getValue(name).(CompileTimeConstantExpr).getStringValue()
+  }
+
+  /**
+   * If the value type of the annotation element with the specified `name` is `int`,
+   * gets the int value used for that element. This includes default values in case no
+   * explicit value is specified.
+   */
+  int getValueInt(string name) {
+    // Uses CompileTimeConstantExpr instead of IntegerLiteral because value can
+    // be read of final variable as well
+    result = getValue(name).(CompileTimeConstantExpr).getIntValue()
+  }
+
+  /**
+   * If the value type of the annotation element with the specified `name` is `boolean`,
+   * gets the boolean value used for that element. This includes default values in case
+   * no explicit value is specified.
+   */
+  boolean getValueBoolean(string name) {
+    // Uses CompileTimeConstantExpr instead of BooleanLiteral because value can
+    // be read of final variable as well
+    result = getValue(name).(CompileTimeConstantExpr).getBooleanValue()
+  }
+
+  /**
+   * If the annotation element with the specified `name` has a Java `Class` as value type,
+   * gets the referenced type used as value for that element. This includes default values
+   * in case no explicit value is specified.
+   */
+  Type getValueClass(string name) { result = getValue(name).(TypeLiteral).getReferencedType() }
```

Comment:
Rename `getReferencedType` for consistency

Reply:
Classes are only one possible value type of annotation type elements, I chose `getValue...` for consistency with the other predicates retrieving an annotation value.

Reply:
Have renamed it to `getTypeValue`.